daemon/server/server.go

```diff
@@ -6,27 +6,35 @@ import (
 
 	"github.com/scootdev/scoot/daemon/protocol"
 	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/snapshot"
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 )
 
 // Create a protocol.ScootDaemonServer
-func NewServer(runner runner.Runner) (*Server, error) {
-	s := &Server{
+func NewServer(runner runner.Runner, filer snapshot.Filer) (Server, error) {
+	s := &server{
 		runner: runner,
+		filer:  filer,
 		server: grpc.NewServer(),
 	}
 	protocol.RegisterScootDaemonServer(s.server, s)
 	return s, nil
 }
 
-type Server struct {
+type Server interface {
+	ListenAndServe() error
+	Stop()
+}
+
+type server struct {
 	runner runner.Runner
+	filer  snapshot.Filer
 
 	server *grpc.Server
```

Comment:
Can we rename some of these please, as this becomes difficult to read. package server, type server, type field server, etc